jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -83,3 +85,59 @@ class SpectrumPerSpaxel(SinglePixelRegion):
     tool_id = 'jdaviz:spectrumperspaxel'
     action_text = 'See spectrum at a single spaxel'
     tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._spectrum_viewer = None
+        self._previous_bounds = None
+        self._mark = None
+        self._data = None
+
+    def _reset_spectrum_viewer_bounds(self):
+        sv_state = self._spectrum_viewer.state
+        sv_state.x_min = self._previous_bounds[0]
+        sv_state.x_max = self._previous_bounds[1]
+        sv_state.y_min = self._previous_bounds[2]
+        sv_state.y_max = self._previous_bounds[3]
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_move, events=['mousemove', 'mouseleave'])
+        if self._spectrum_viewer is None:
+            self._spectrum_viewer = self.viewer.jdaviz_helper.app.get_viewer('spectrum-viewer')
```

Comment:
can we avoid hardcoding `'spectrum-viewer'` here by using the helper attributes?

Reply:
Done, grabs the first profile viewer now.